src/core/Akka/Dispatch/CachingConfig.cs

```diff
@@ -0,0 +1,277 @@
+﻿/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ * Original C# code written by Akka.NET project <http://getakka.net/>
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Akka.Configuration;
+using Akka.Configuration.Hocon;
+
+namespace Akka.Dispatch
+{
+    /// <summary>
+    /// INTERNAL API
+    /// 
+    /// A <see cref="CachingConfig"/> is a <see cref="Config"/> that wraps another <see cref="Config"/> and is used to
+    /// cache path lookup and string retrieval, which we happen to do in some critical paths of the actor creation
+    /// and mailbox selection code.
+    /// 
+    /// All other <see cref="Config"/> operations are delegated to the wrapped <see cref="Config"/>.
+    /// </summary>
+    class CachingConfig : Config
```

Comment:
This should be used carefully as it could interfere with `InjectTopLevelFallback`. where the config might cache a value that is later overridden by a submodule, e.g. remoting/persistence.

Is this handled in the current impl? is this recreated when new fallbacks are applied?